erpnext/templates/includes/integrations/authorizenet_checkout.js

```diff
@@ -52,3 +52,14 @@ $('input[data-validation="digit"]')
 	.keypress(function(event) {
 		return (event.charCode !== 8 && event.charCode === 0 || (event.charCode >= 48 && event.charCode <= 57));
 	});
+
+	$('#card-number').on('keyup', function (e) {
+    var val = $(this).val();
+    var newval = '';
+    val = val.replace(/\s/g, '');
+    for (var i = 0; i < val.length; i++) {
+      if (i % 4 == 0 && i > 0) newval = newval.concat(' ');
+      newval = newval.concat(val[i]);
+    }
+    $(this).val(newval);
```

Comment:
```suggestion
        var val = $(this).val();
        var newval = '';
        val = val.replace(/\s/g, '');
        for (var i = 0; i < val.length; i++) {
          if (i % 4 == 0 && i > 0) newval = newval.concat(' ');
          newval = newval.concat(val[i]);
        }
        $(this).val(newval);
```

Also did you make changes when the card number is fetched for processing ?? Wont there be spaces in that ?

Reply:
Hey @hrwX I made changes to jquery to remove the spaces while processing and it seems to work fine on my local